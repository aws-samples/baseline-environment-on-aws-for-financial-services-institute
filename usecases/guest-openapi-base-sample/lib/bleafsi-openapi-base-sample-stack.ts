import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { aws_apigateway as apigw } from 'aws-cdk-lib';
import { Cognito } from './cognito';
import { ApiGw } from './api-gw';
import { StackParameter } from 'bin/parameter';
import { CloudFront } from './cloudfront';
import { WafV2ApiGW } from './waf-apigw';

/**
 * OpenAPI basic ワークロード サンプルアプリケーションを作成するStack
 */
export class SampleOpenApiStack extends cdk.Stack {
  public readonly sample_api: apigw.RestApi;

  constructor(scope: Construct, id: string, props: StackParameter) {
    super(scope, id, props);

    //Cognito UserPooL作成
    const cognito = new Cognito(this, 'Cognito Pool');

    //API Gateway作成
    const apiGw = new ApiGw(this, 'ApiGateway', cognito, props);

    //Waf v2 for API Gateway作成
    const wafv2 = new WafV2ApiGW(this, 'WafForApiGW', apiGw.sampleApi.restApiId);
    //デプロイ時の依存関係を設定
    wafv2.webAclApigw.node.addDependency(apiGw.sampleApi);

    //CloudFront distribution作成
    new CloudFront(this, 'CloudFront', props);
  }
}
