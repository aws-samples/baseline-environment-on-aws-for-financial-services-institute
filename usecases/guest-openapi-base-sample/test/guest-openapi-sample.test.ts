import * as cdk from 'aws-cdk-lib';
import { Template, Annotations, Match } from 'aws-cdk-lib/assertions';
import { AwsSolutionsChecks, NagSuppressions } from 'cdk-nag';
import { SampleOpenApiStack } from '../lib/bleafsi-openapi-base-sample-stack';
import { DevParameter } from '../bin/parameter';

const app = new cdk.App({
  context: {
    pjPrefix: 'BLEA-FSI',
    customdomainname: 'api.xxx.xxx',
    alterdomainname: 'openapi-base.xxx.xxx',
    'certIdarn-apigw': 'arn:aws:acm:ap-northeast-1:xxxxxxxxxxxx:certificate/xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
    'certIdarn-cf': 'arn:aws:acm:us-east-1:xxxxxxxxxxxx:certificate/xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  },
});

let apiApp: SampleOpenApiStack;

/*const appProps: OpenApiBaseContextProps = {
  customdomainName: 'api.xxx.xxx',
  alterdomainName: 'openapi-base.xxx.xxx',
  certIdarnApigw: 'arn:aws:acm:ap-northeast-1:xxxxxxxxxxxx:certificate/xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  certIdarnCf: 'arn:aws:acm:us-east-1:xxxxxxxxxxxx:certificate/xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  env: {
    account: process.env.CDK_DEFAULT_ACCOUNT,
    region: process.env.CDK_DEFAULT_REGION,
  },
};*/
describe(`BLEA-FSI snapshot check`, () => {
  test('OpenAPI sample Stacks', () => {
    apiApp = new SampleOpenApiStack(app, 'SampleOpenApiStack', DevParameter);

    // test with snapshot
    expect(Template.fromStack(apiApp)).toMatchSnapshot();
  });
});

// cdk-nag suppression
describe(`BLEA-FSI cdk-nag AwsSolutions Pack: api`, () => {
  beforeAll(() => {
    NagSuppressions.addResourceSuppressionsByPath(
      apiApp,
      '/SampleOpenApiStack/Cognito Pool/userPool/smsRole/Resource',
      [{ id: 'AwsSolutions-IAM5', reason: 'It is used only when deploying.' }],
    );
    NagSuppressions.addResourceSuppressionsByPath(apiApp, '/SampleOpenApiStack/Cognito Pool/userPool/Resource', [
      { id: 'AwsSolutions-COG2', reason: 'It is a sample build project.' },
    ]);
    NagSuppressions.addResourceSuppressionsByPath(apiApp, '/SampleOpenApiStack/Cognito Pool/userPool/Resource', [
      { id: 'AwsSolutions-COG3', reason: 'It is a sample build project.' },
    ]);
    NagSuppressions.addResourceSuppressionsByPath(
      apiApp,
      '/SampleOpenApiStack/Cognito Pool/HelloHandler/ServiceRole/Resource',
      [{ id: 'AwsSolutions-IAM4', reason: 'It is used only when deploying.' }],
    );
    NagSuppressions.addResourceSuppressionsByPath(apiApp, '/SampleOpenApiStack/ApiGateway/sample-api/Resource', [
      { id: 'AwsSolutions-APIG2', reason: 'It is a sample build project.' },
    ]);
    NagSuppressions.addResourceSuppressionsByPath(
      apiApp,
      '/SampleOpenApiStack/ApiGateway/sample-api/CloudWatchRole/Resource',
      [{ id: 'AwsSolutions-IAM4', reason: 'It is a sample build project.' }],
    );
    NagSuppressions.addResourceSuppressionsByPath(
      apiApp,
      '/SampleOpenApiStack/ApiGateway/sample-api/DeploymentStage.prod/Resource',
      [{ id: 'AwsSolutions-APIG3', reason: 'cdk-nag tool bug.' }],
    );
    NagSuppressions.addResourceSuppressionsByPath(apiApp, '/SampleOpenApiStack/CloudFront/AppLogBucket/Resource', [
      { id: 'AwsSolutions-S1', reason: 'It is a log bucket.' },
    ]);

    cdk.Aspects.of(apiApp).add(new AwsSolutionsChecks());
  });

  test('No unsuppressed Errors', () => {
    const errors = Annotations.fromStack(apiApp).findError('*', Match.stringLikeRegexp('AwsSolutions-.*'));
    try {
      expect(errors).toHaveLength(0);
      console.log('cdk-nag: no errors for stack ' + apiApp.stackName);
    } catch (e) {
      const errorMessages = errors.map((e) => ({
        type: e.entry.type,
        data: e.entry.data,
        id: e.id,
      }));
      console.error(JSON.stringify(errorMessages, undefined, 2));
      throw e;
    }
  });
});
