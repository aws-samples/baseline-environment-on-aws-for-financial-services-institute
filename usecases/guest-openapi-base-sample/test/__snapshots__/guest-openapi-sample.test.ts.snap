// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BLEA-FSI snapshot check OpenAPI sample Stacks 1`] = `
Object {
  "Outputs": Object {
    "ExportsOutputRefsampleapiAA5ED7E46021F1DE": Object {
      "Export": Object {
        "Name": "SampleOpenApiStack:ExportsOutputRefsampleapiAA5ED7E46021F1DE",
      },
      "Value": Object {
        "Ref": "sampleapiAA5ED7E4",
      },
    },
    "sampleapiEndpoint5691AF90": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "sampleapiAA5ED7E4",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "sampleapiDeploymentStageprod1B60AC82",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "HelloHandler2E4FBA4D": Object {
      "DependsOn": Array [
        "HelloHandlerServiceRole11EF7C63",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "222a3a7cad2a200ab801e4bfe3ce4a5e3d04a8d24140b25045c4de9cb03562f8.zip",
        },
        "Handler": "hello.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "HelloHandlerServiceRole11EF7C63",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "HelloHandlerServiceRole11EF7C63": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "OpenApiCustomDomain61CD031B": Object {
      "Properties": Object {
        "DomainName": "api.xxx.xxx",
        "EndpointConfiguration": Object {
          "Types": Array [
            "REGIONAL",
          ],
        },
        "RegionalCertificateArn": "arn:aws:acm:ap-northeast-1:xxxxxxxxxxxx:certificate/xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
      },
      "Type": "AWS::ApiGateway::DomainName",
    },
    "OpenApiCustomDomainMapSampleOpenApiStacksampleapi9BE1FF5B2AA27007": Object {
      "Properties": Object {
        "DomainName": Object {
          "Ref": "OpenApiCustomDomain61CD031B",
        },
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
        "Stage": Object {
          "Ref": "sampleapiDeploymentStageprod1B60AC82",
        },
      },
      "Type": "AWS::ApiGateway::BasePathMapping",
    },
    "RestApiLogAccessLogGroup9DDCF09E": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": "/aws/apigateway/rest-api-access-log",
        "RetentionInDays": 365,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "cognitoAuthorizer9878DCCA": Object {
      "Properties": Object {
        "IdentitySource": "method.request.header.Authorization",
        "Name": "CognitoAuthorizer",
        "ProviderARNs": Array [
          Object {
            "Fn::GetAtt": Array [
              "userPoolDC9497E0",
              "Arn",
            ],
          },
        ],
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
        "Type": "COGNITO_USER_POOLS",
      },
      "Type": "AWS::ApiGateway::Authorizer",
    },
    "sampleapiAA5ED7E4": Object {
      "Properties": Object {
        "Name": "sample-api",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "sampleapiAccount074F5E10": Object {
      "DeletionPolicy": "Retain",
      "DependsOn": Array [
        "sampleapiAA5ED7E4",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "sampleapiCloudWatchRole14C66026",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "sampleapiCloudWatchRole14C66026": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "sampleapiDeployment9D4ABE32102e226748b676046673f976fa391f7a": Object {
      "DependsOn": Array [
        "sampleapiproxyGETBD44B935",
        "sampleapiproxyDA278B45",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "sampleapiDeploymentStageprod1B60AC82": Object {
      "DependsOn": Array [
        "sampleapiAccount074F5E10",
      ],
      "Properties": Object {
        "AccessLogSetting": Object {
          "DestinationArn": Object {
            "Fn::GetAtt": Array [
              "RestApiLogAccessLogGroup9DDCF09E",
              "Arn",
            ],
          },
          "Format": "$context.identity.sourceIp $context.identity.caller $context.identity.user [$context.requestTime] \\"$context.httpMethod $context.resourcePath $context.protocol\\" $context.status $context.responseLength $context.requestId",
        },
        "DeploymentId": Object {
          "Ref": "sampleapiDeployment9D4ABE32102e226748b676046673f976fa391f7a",
        },
        "MethodSettings": Array [
          Object {
            "DataTraceEnabled": true,
            "HttpMethod": "*",
            "LoggingLevel": "INFO",
            "ResourcePath": "/*",
          },
        ],
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "sampleapiproxyDA278B45": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "sampleapiAA5ED7E4",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "sampleapiproxyGETApiPermissionSampleOpenApiStacksampleapi9BE1FF5BGETproxyC62D7D08": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "sampleapiAA5ED7E4",
              },
              "/",
              Object {
                "Ref": "sampleapiDeploymentStageprod1B60AC82",
              },
              "/GET/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "sampleapiproxyGETApiPermissionTestSampleOpenApiStacksampleapi9BE1FF5BGETproxy3BF36196": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "sampleapiAA5ED7E4",
              },
              "/test-invoke-stage/GET/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "sampleapiproxyGETBD44B935": Object {
      "Properties": Object {
        "AuthorizationType": "COGNITO_USER_POOLS",
        "AuthorizerId": Object {
          "Ref": "cognitoAuthorizer9878DCCA",
        },
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "HelloHandler2E4FBA4D",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "sampleapiproxyDA278B45",
        },
        "RestApiId": Object {
          "Ref": "sampleapiAA5ED7E4",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "userPoolDC9497E0": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AccountRecoverySetting": Object {
          "RecoveryMechanisms": Array [
            Object {
              "Name": "verified_email",
              "Priority": 1,
            },
          ],
        },
        "AdminCreateUserConfig": Object {
          "AllowAdminCreateUserOnly": true,
        },
        "AutoVerifiedAttributes": Array [
          "email",
        ],
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "EnabledMfas": Array [
          "SMS_MFA",
        ],
        "MfaConfiguration": "OPTIONAL",
        "Policies": Object {
          "PasswordPolicy": Object {
            "MinimumLength": 8,
            "RequireLowercase": true,
            "RequireNumbers": true,
            "RequireSymbols": true,
            "RequireUppercase": true,
            "TemporaryPasswordValidityDays": 7,
          },
        },
        "Schema": Array [
          Object {
            "Mutable": true,
            "Name": "email",
            "Required": true,
          },
          Object {
            "Mutable": true,
            "Name": "phone_number",
            "Required": true,
          },
        ],
        "SmsConfiguration": Object {
          "ExternalId": "SampleOpenApiStackuserPoolF20B64B3",
          "SnsCallerArn": Object {
            "Fn::GetAtt": Array [
              "userPoolsmsRoleED3E1FD9",
              "Arn",
            ],
          },
        },
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "UserPoolName": "apiUserPool",
        "UsernameAttributes": Array [
          "email",
        ],
        "UsernameConfiguration": Object {
          "CaseSensitive": false,
        },
        "VerificationMessageTemplate": Object {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Delete",
    },
    "userPoolclient992D31C6": Object {
      "Properties": Object {
        "AllowedOAuthFlows": Array [
          "implicit",
          "code",
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": Array [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin",
        ],
        "CallbackURLs": Array [
          "https://example.com",
        ],
        "ClientName": "apiUserPoolClient",
        "ExplicitAuthFlows": Array [
          "ALLOW_USER_PASSWORD_AUTH",
          "ALLOW_ADMIN_USER_PASSWORD_AUTH",
          "ALLOW_USER_SRP_AUTH",
          "ALLOW_REFRESH_TOKEN_AUTH",
        ],
        "GenerateSecret": false,
        "SupportedIdentityProviders": Array [
          "COGNITO",
        ],
        "UserPoolId": Object {
          "Ref": "userPoolDC9497E0",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
    "userPoolsmsRoleED3E1FD9": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Condition": Object {
                "StringEquals": Object {
                  "sts:ExternalId": "SampleOpenApiStackuserPoolF20B64B3",
                },
              },
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cognito-idp.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "sns:Publish",
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "sns-publish",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`BLEA-FSI snapshot check OpenAPI sample Stacks 2`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "SampleWafAclApigw": Object {
      "Properties": Object {
        "DefaultAction": Object {
          "Allow": Object {},
        },
        "Name": "sample-waf-web-acl-apigw",
        "Rules": Array [
          Object {
            "Name": "AWSManagedRulesCommonRuleSet",
            "OverrideAction": Object {
              "Count": Object {},
            },
            "Priority": 1,
            "Statement": Object {
              "ManagedRuleGroupStatement": Object {
                "Name": "AWSManagedRulesCommonRuleSet",
                "VendorName": "AWS",
              },
            },
            "VisibilityConfig": Object {
              "CloudWatchMetricsEnabled": true,
              "MetricName": "AWS-AWSManagedRulesCommonRuleSet",
              "SampledRequestsEnabled": true,
            },
          },
          Object {
            "Name": "AWSManagedRulesKnownBadInputsRuleSet",
            "OverrideAction": Object {
              "Count": Object {},
            },
            "Priority": 2,
            "Statement": Object {
              "ManagedRuleGroupStatement": Object {
                "Name": "AWSManagedRulesKnownBadInputsRuleSet",
                "VendorName": "AWS",
              },
            },
            "VisibilityConfig": Object {
              "CloudWatchMetricsEnabled": true,
              "MetricName": "AWS-AWSManagedRulesKnownBadInputsRuleSet",
              "SampledRequestsEnabled": true,
            },
          },
          Object {
            "Name": "AWSManagedRulesAmazonIpReputationList",
            "OverrideAction": Object {
              "Count": Object {},
            },
            "Priority": 3,
            "Statement": Object {
              "ManagedRuleGroupStatement": Object {
                "Name": "AWSManagedRulesAmazonIpReputationList",
                "VendorName": "AWS",
              },
            },
            "VisibilityConfig": Object {
              "CloudWatchMetricsEnabled": true,
              "MetricName": "AWS-AWSManagedRulesAmazonIpReputationList",
              "SampledRequestsEnabled": true,
            },
          },
          Object {
            "Name": "AWSManagedRulesLinuxRuleSet",
            "OverrideAction": Object {
              "Count": Object {},
            },
            "Priority": 4,
            "Statement": Object {
              "ManagedRuleGroupStatement": Object {
                "Name": "AWSManagedRulesLinuxRuleSet",
                "VendorName": "AWS",
              },
            },
            "VisibilityConfig": Object {
              "CloudWatchMetricsEnabled": true,
              "MetricName": "AWS-AWSManagedRulesLinuxRuleSet",
              "SampledRequestsEnabled": true,
            },
          },
          Object {
            "Name": "AWSManagedRulesSQLiRuleSet",
            "OverrideAction": Object {
              "Count": Object {},
            },
            "Priority": 5,
            "Statement": Object {
              "ManagedRuleGroupStatement": Object {
                "Name": "AWSManagedRulesSQLiRuleSet",
                "VendorName": "AWS",
              },
            },
            "VisibilityConfig": Object {
              "CloudWatchMetricsEnabled": true,
              "MetricName": "AWS-AWSManagedRulesSQLiRuleSet",
              "SampledRequestsEnabled": true,
            },
          },
        ],
        "Scope": "REGIONAL",
        "VisibilityConfig": Object {
          "CloudWatchMetricsEnabled": true,
          "MetricName": "sample-waf-web-acl",
          "SampledRequestsEnabled": true,
        },
      },
      "Type": "AWS::WAFv2::WebACL",
    },
    "TestWebAclAssociationApigw": Object {
      "Properties": Object {
        "ResourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:aws:apigateway:ap-northeast-1::/restapis/",
              Object {
                "Fn::ImportValue": "SampleOpenApiStack:ExportsOutputRefsampleapiAA5ED7E46021F1DE",
              },
              "/stages/prod",
            ],
          ],
        },
        "WebACLArn": Object {
          "Fn::GetAtt": Array [
            "SampleWafAclApigw",
            "Arn",
          ],
        },
      },
      "Type": "AWS::WAFv2::WebACLAssociation",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`BLEA-FSI snapshot check OpenAPI sample Stacks 3`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "AppLogBucket71C94073": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AccessControl": "LogDeliveryWrite",
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "app-logs-cloudfront",
        "OwnershipControls": Object {
          "Rules": Array [
            Object {
              "ObjectOwnership": "ObjectWriter",
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "AppLogBucketPolicyF1357798": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "AppLogBucket71C94073",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "AppLogBucket71C94073",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "AppLogBucket71C94073",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "s3:*",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cloudfront.amazonaws.com",
              },
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::",
                      Object {
                        "Ref": "AppLogBucket71C94073",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::",
                      Object {
                        "Ref": "AppLogBucket71C94073",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
              "Sid": "AllowCloudFrontOnly",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "OpenApiDistributionDD71B494": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "Aliases": Array [
            "openapi-base.xxx.xxx",
          ],
          "Comment": "OpenAPI-distribution",
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachePolicyId": Object {
              "Ref": "cachePolicyForApiGatewayFE08C17E",
            },
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "TargetOriginId": "SampleCfStackOpenApiDistributionOrigin17F9DAD35",
            "ViewerProtocolPolicy": "https-only",
          },
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Logging": Object {
            "Bucket": Object {
              "Fn::GetAtt": Array [
                "AppLogBucket71C94073",
                "RegionalDomainName",
              ],
            },
          },
          "Origins": Array [
            Object {
              "CustomOriginConfig": Object {
                "OriginProtocolPolicy": "https-only",
                "OriginSSLProtocols": Array [
                  "TLSv1.2",
                ],
              },
              "DomainName": "api.xxx.xxx",
              "Id": "SampleCfStackOpenApiDistributionOrigin17F9DAD35",
            },
          ],
          "PriceClass": "PriceClass_All",
          "Restrictions": Object {
            "GeoRestriction": Object {
              "Locations": Array [
                "US",
                "GB",
                "JP",
              ],
              "RestrictionType": "whitelist",
            },
          },
          "ViewerCertificate": Object {
            "AcmCertificateArn": "arn:aws:acm:us-east-1:xxxxxxxxxxxx:certificate/xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "cachePolicyForApiGatewayFE08C17E": Object {
      "Properties": Object {
        "CachePolicyConfig": Object {
          "DefaultTTL": 86400,
          "MaxTTL": 31536000,
          "MinTTL": 0,
          "Name": "cachePolicyForApiGateway",
          "ParametersInCacheKeyAndForwardedToOrigin": Object {
            "CookiesConfig": Object {
              "CookieBehavior": "none",
            },
            "EnableAcceptEncodingBrotli": true,
            "EnableAcceptEncodingGzip": true,
            "HeadersConfig": Object {
              "HeaderBehavior": "whitelist",
              "Headers": Array [
                "Authorization",
              ],
            },
            "QueryStringsConfig": Object {
              "QueryStringBehavior": "none",
            },
          },
        },
      },
      "Type": "AWS::CloudFront::CachePolicy",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
